### PR TITLE
More mira port

### LINKED ISSUE #245: Port new CIME to Mira for ACME
The initial port of new CIME to Mira was done as part of Issue #205 / PR #240. However, although the new CIME works on Mira for CESM, it does not work (setup fails) for ACME (CIME_MODEL=acme).

### PR DESCRIPTION
**Pull request by jedwards4b**
This change gets a basic case setup working on mira again.  However the workflow - running st_archive and then resubmitting does not yet work.   This means that ERR and ERI tests will not work.  mpi-serial is also not supported on this platform so most of the scripts_regression_tests will also fail.

Also including updates (machine file updates) for machines rarely used/tested - Updating batch system type for babbage, brutus, eos, erebus, hera, pleiades-wes, sierra, titan.

Also build output is now logged, streaming it to a file, instead of buffering it in memory.

Test suite: 
Test baseline: 
Test namelist changes: 
Test status: [bit for bit, roundoff, climate changing]

Closes #245 
User interface changes?: 

Code review: jayesh

### CODE REVIEW

**Comment by jayeshkrishna** [comment]
I am going to try testing this change using CIME_MODEL=cesm with the following case,
```
./create_newcase -case cime_sanity_check_mira -compset X -res f19_g16
```

**Comment by jayeshkrishna** [comment]
This PR also seems to have changes for other (not mira) machines. Is that intentional?

File: cime_config/cesm/machines/config_batch.xml [hunk 1]
```diff
@@ -142,21 +143,21 @@
    </batch_system>
 
   <!-- babbage is PBS -->
-  <batch_system MACH="babbage" version="x.y">
```
Comment (jedwards4b) [inline cime_config/cesm/machines/config_batch.xml]: I updated some config_batch.xml for machines that are rarely used or tested.

File: utils/python/CIME/build.py [hunk 2]
```diff
@@ -94,10 +93,11 @@ def build_model(case, build_threaded, exeroot, clm_config_opts, incroot, complis
     file_build = os.path.join(exeroot, "%s.bldlog.%s" % (cime_model, lid))
 
     config_dir = os.path.join(cimeroot, "driver_cpl", "cime_config")
-    stat = run_cmd("%s/buildexe %s %s %s >> %s 2>&1" %
-                   (config_dir, caseroot, bldroot, libroot, file_build),
-                   from_dir=bldroot, ok_to_fail=True, verbose=True)[0]
-
+    f = open(file_build, "w")
```
Comment (jedwards4b) [inline utils/python/CIME/build.py]: This changes the way that build output is logged, streaming it to the file instead of buffering it in memory.

**Comment by jayeshkrishna** [comment]
@jedwards4b : I updated the PR message with the information (changes not related to mira) above. Please go through it and feel free to update/modify the new comments.

**Comment by jedwards4b** [comment]
looks good, thank you

**Comment by jayeshkrishna** [comment]
I have submitted a job on Cetus for CIME_MODEL=cesm (create/setup/build/submit succeeds).

The ./create_newcase command failed with CIME_MODEL=acme . This will need to be pursued as part of Issue #245 . The output from create_newcase (CIME_MODEL=acme) is shown below,

```
[jayesh@miralac1 scripts (master>)]$ ./create_newcase -case acme_sanity_check_mira -compset X -res f19_g16                       
Compset longname is 2000_XATM_XLND_XICE_XOCN_XROF_XGLC_XWAV 
Compset specification file is /gpfs/mira-home/jayesh/acme/ESMCI_cime_merge/scripts/Tools/../../driver_cpl/cime_config/config_compsets.xml
Pes     specification file is /gpfs/mira-home/jayesh/acme/ESMCI_cime_merge/scripts/Tools/../../driver_cpl/cime_config/config_pes.xml
Pes setting: grid          is a%1.9x2.5_l%1.9x2.5_oi%gx1v6_r%r05_m%gx1v6_g%null_w%null 
Pes setting: compset       is 2000_XATM_XLND_XICE_XOCN_XROF_XGLC_XWAV 
Pes setting: grid match    is any 
Pes setting: machine match is mira 
Pes setting: compset_match is [DX]ATM 
Pes setting: pesize match  is any 
ERROR: Command: '/usr/bin/xmllint --format --output /gpfs/mira-home/jayesh/acme/ESMCI_cime_merge/scripts/acme_sanity_check_mira/env_mach_pes.xml -' failed. See terminal output

```

**Comment by jedwards4b** [comment]
The PR has been updated with a fix for the above, please try again.

**Comment by jayeshkrishna** [comment]
Got the following error with CIME_MODEL=cesm (in the job error output),
```
Not able to fully resolve item '/usr/bin/runjob   $LOCARGS  --envs OMP_STACKSIZE=64M  --envs OMP_NUM_THREADS=8  --envs BG_THREADLAYOUT=1  --label short  -p 8  -n 8 :  /projects/ClimateEnergy/usr/jayesh/cesm_sanity_check_mira/bld/acme.exe  >> acme.log.160714-213314 2>&1 '
ERROR: Command: '/usr/bin/runjob   $LOCARGS  --envs OMP_STACKSIZE=64M  --envs OMP_NUM_THREADS=8  --envs BG_THREADLAYOUT=1  --label short  -p 8  -n 8 :  /projects/ClimateEnergy/usr/jayesh/cesm_sanity_check_mira/bld/acme.exe  >> acme.log.160714-213314 2>&1 ' failed from dir '/projects/ClimateEnergy/usr/jayesh/cesm_sanity_check_mira/run'. See terminal output

```
@jedwards4b : Can you build and run any case on Mira/Cetus with CIME_MODEL=cesm ? I could try the case that you are using in your tests to see if the fix works.

**Comment by jayeshkrishna** [comment]
I have submitted a job (case creates+builds+submits correctly) on Mira for CIME_MODEL=cesm . 
This PR still does not fix issues for CIME_MODEL=acme. I can work on that since CIME_MODEL=cesm is working now.

**Comment by jedwards4b** [comment]
I'm sorry I thought that it would work for acme - what problem are you seeing?

**Comment by jayeshkrishna** [comment]
No worries, I will start looking at the issues (the first one was due to missing PES_PER_NODE tag). Let me try to get it working and will let you know the details.

**Comment by jayeshkrishna** [comment]
I got the following error at runtime (CIME_MODEL=cesm) after merging the branch locally to master,
```
Not able to fully resolve item '/usr/bin/runjob   $LOCARGS  --envs OMP_STACKSIZE=64M  --envs OMP_NUM_THREADS=8  --envs BG_THREADLAYOUT=1  --label short  -p 8  -n 8 :  /projects/ClimateEnergy/usr/jayesh/cesm_sanity_check_mira/bld/acme.exe  >> acme.log.160719-210632 2>&1 '
ERROR: Command: '/usr/bin/runjob   $LOCARGS  --envs OMP_STACKSIZE=64M  --envs OMP_NUM_THREADS=8  --envs BG_THREADLAYOUT=1  --label short  -p 8  -n 8 :  /projects/ClimateEnergy/usr/jayesh/cesm_sanity_check_mira/bld/acme.exe  >> acme.log.160719-210632 2>&1 ' failed from dir '/projects/ClimateEnergy/usr/jayesh/cesm_sanity_check_mira/run'. See terminal output

```
Will try it with master and see how it goes.

**Comment by jedwards4b** [comment]
The "not able to fully resolve" is not an error, it is expected on mira because of the LOCARGS flag.    
Look in acme.log.160719-210632 for the real problem.

**Comment by jayeshkrishna** [comment]
The error message above is with CIME_MODEL=cesm, and I only have <JOBID>.{output|error|cobaltlog} in the case directory

**Comment by jedwards4b** [comment]
the acme.log file will be in the run directory

**Comment by jayeshkrishna** [comment]
Looks like an issue in my environment, although I had set CIME_MODEL=cesm while building the model my shell startup files had CIME_MODEL=acme. After changing it to CIME_MODEL=cesm, the job ran successfully.
Going to try the ACME side now.

**Comment by rljacob** [comment]
Did this also fix #253 ?

**Comment by jedwards4b** [comment]
I believe so

**Pull request merged**